main.tf

```diff
@@ -41,6 +41,18 @@ resource "aws_dynamodb_table" "basic-dynamodb-table" {
     }
   }
 
+  dynamic "attribute" {
+    for_each = [
+      for i in var.global_secondary_indeces : i
+      if can(i.range_key)
```

Comment:
maybe we need to validate range key type as well ?

Reply:
In the current condition, a missing range_key_type will cause the apply to fail when creating the attribute.
If I add the range_key_type check, apply will still fail in creating the index because the attribute was not created.
I think that it will be easier to debug an error saying that range_key_type is missing.